### PR TITLE
update river and waylock

### PR DESCRIPTION
**Pull request by dkwo**
- I tested the changes in this PR: yes. been running them for months, built with zig-static
- I built this PR locally for my native architecture, (x86_64-musl)

### CODE REVIEW

**Comment by icp1994** [comment]
oof...building LLVM/Zig in CI

**Comment by icp1994** [comment]
I checked with the local template for `river` I had, noticed these two differences
* no `-Drelease-safe` (it's enforced by the [zig build style](https://github.com/void-linux/void-packages/blob/master/common/build-style/zig-build.sh))
* `libxkbcommon-devel` & `wayland-devel` in makdepends (I read somewhere wayland-devel is needed in both host & target sometimes)

**Comment by dkwo** [comment]
I wonder if by now it is acceptable to drop `build_options_default="xwayland"`..

File: srcpkgs/river/template [hunk 1]
```diff
@@ -1,21 +1,21 @@
 # Template file for 'river'
 pkgname=river
-version=0.2.3
+version=0.2.4
 revision=1
 build_style=zig-build
-configure_args="$(vopt_if xwayland -Dxwayland) -Dpie"
+configure_args="$(vopt_if xwayland -Dxwayland) -Dpie -Dstrip"
```
Comment (ifreund) [inline srcpkgs/river/template:6]: -Dstrip will break the river-dbg package

File: srcpkgs/river/template [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
 maintainer="Isaac Freund <mail@isaacfreund.com>"
 license="GPL-3.0-or-later"
 homepage="https://github.com/riverwm/river"
-distfiles="${homepage}/releases/download/v${version}/river-${version}.tar.gz"
-checksum=7acdd2b929e16a20006b4baf1f29e7a8dc7f70cf5747e1503a610a90d1637132
+distfiles="https://github.com/riverwm/river/releases/download/v${version}/river-${version}.tar.gz"
+checksum=26c1c41a65ce3804069afad6988410515cf478d2b76303ebc699766d3d4dc69f
 
 build_options="xwayland"
-build_options_default="xwayland"
```
Comment (ifreund) [inline srcpkgs/river/template:18]: Why change this default? Most users will want Xwayland likely and those that know what it is and know they don't want it shouldn't have much trouble disabling it.

File: srcpkgs/waylock/template [hunk 3]
```diff
@@ -1,16 +1,17 @@
 # Template file for 'waylock'
 pkgname=waylock
-version=0.3.5
+version=0.6.2
 revision=1
-build_style=cargo
-hostmakedepends="pkg-config"
-makedepends="libxkbcommon-devel pam-devel"
+build_style=zig-build
+configure_args="-Dpie -Dstrip"
```
Comment (ifreund) [inline srcpkgs/waylock/template:6]: Again, -Dstrip will break waylock-dbg

File: srcpkgs/waylock/template [hunk 4]
```diff
@@ -1,16 +1,17 @@
...
+hostmakedepends="pkg-config wayland-devel"
+makedepends="libxkbcommon-devel pam-devel wayland-protocols wayland-devel scdoc"
```
Comment (ifreund) [inline srcpkgs/waylock/template:8]: Should scdoc not be in hostmakedepends?

File: srcpkgs/river/template [hunk 5]
```diff
@@ -1,21 +1,21 @@
...
-configure_args="$(vopt_if xwayland -Dxwayland) -Dpie"
+configure_args="$(vopt_if xwayland -Dxwayland) -Dpie -Dstrip"
 hostmakedepends="pkg-config wayland-devel scdoc"
-makedepends="wlroots-devel libevdev-devel"
+makedepends="wlroots-devel libevdev-devel pixman-devel
+ wayland-protocols libxkbcommon-devel wayland-devel"
 depends="$(vopt_if xwayland xorg-server-xwayland)"
 short_desc="Dynamic tiling Wayland compositor"
 maintainer="Isaac Freund <mail@isaacfreund.com>"
 license="GPL-3.0-or-later"
```
Comment (ifreund) [inline srcpkgs/river/template:13]: This has changed to GPL-3.0-only

**Review by ifreund — COMMENTED** [review]
Crazy that void finally has LLVM 15!

Thanks for updating these :)

**Comment by classabbyamp** [comment]
make sure to rebase and remove the broken= in river, that was temporary to get the builders to finish with llvm

**Comment by ifreund** [comment]
(The broken= was added due to build issues on 32 bit arm targets, which should likely be disabled)

File: srcpkgs/river/template (hunk 1 above)
Comment (dkwo) [reply srcpkgs/river/template]: done

File: srcpkgs/river/template (hunk 2 above)
Comment (dkwo) [reply srcpkgs/river/template:18]: i thought pure wayland could be the default by now..

File: srcpkgs/waylock/template (hunk 3 above)
Comment (dkwo) [reply srcpkgs/waylock/template]: done

File: srcpkgs/river/template (hunk 5 above)
Comment (dkwo) [reply srcpkgs/river/template]: done

**Comment by dkwo** [comment]
should i restrict arch? maybe `~armarmv6l* ~armv7l* ~i686*` ?

File: srcpkgs/waylock/template (hunk 4 above)
Comment (dkwo) [reply srcpkgs/waylock/template]: done

File: srcpkgs/river/template (hunk 2 above)
Comment (paper42) [reply srcpkgs/river/template:18]: I am not sure I understand why people wouldn't want xwayland support in their compositor.

**Comment by classabbyamp** [comment]
> should i restrict arch? maybe `~armv6l* ~armv7l* ~i686*` ?

maybe replace the conditional around the broken with `"$XBPS_TARGET_WORD_SIZE" = 32`

**Comment by icp1994** [comment]
I thought since Zig template has `archs="x86_64* aarch64*"`, river would have that implied. Isn't that how electron packages work too? Can someone ELI5 a bit?

**Comment by classabbyamp** [comment]
it's hostmakedepends

File: srcpkgs/river/template (hunk 2 above)
Comment (dkwo) [reply srcpkgs/river/template:18]: since wayland support is more mature, one could argue that having xwayland enabled is less secure and not worth it.

Comment (dkwo) [reply srcpkgs/river/template:18]: for example, i don't need to worry about https://github.com/void-linux/void-packages/issues/43174

**Comment by dkwo** [comment]
is the repository syncing correctly by now, or should i wait to rebase this?

**Comment by paper42** [comment]
> is the repository syncing correctly by now, or should i wait to rebase this?

CI repository was switched to repo-de for now, so it will work

**Comment by dkwo** [comment]
ready to merge imho
